packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx

```diff
@@ -0,0 +1,180 @@
+      z.infer<typeof CommunityAlert>
+    >) || [],
+  );
+
+  const { bundledSubs } = useNotificationSettings();
+
+  const [threadsFilter, setThreadsFilter] = useState<readonly number[]>([]);
+  const [section, setSection] =
+    useState<NotificationSection>('community-alerts');
+
+  const handleUnsubscribe = useCallback(
+    (id: number) => {
+      setThreadsFilter([...threadsFilter, id]);
+    },
+    [threadsFilter],
+  );
+
+  const handlePushNotificationSubscription = useCallback(() => {
+    async function doAsync() {
+      const permission = await Notification.requestPermission();
+      if (permission === 'granted') {
+        console.log('Notification permission granted.');
+        const token = await getFirebaseMessagingToken();
+        await registerClientRegistrationToken.mutateAsync({
+          id: 'none',
+          token,
+        });
+      }
+    }
+
+    doAsync().catch(console.error);
+  }, [registerClientRegistrationToken]);
+
+  if (threadSubscriptions.isLoading) {
+    return <PageLoading />;
+  } else if (!app.isLoggedIn()) {
+    navigate('/', { replace: true });
+    return <PageLoading />;
+  }
+
+  return (
+    <CWPageLayout>
+      <div className="NotificationSettingsPage NotificationSettings">
+        <CWText type="h3" fontWeight="semiBold" className="page-header-text">
+          Notification settings
+        </CWText>
+
+        <CWText className="page-subheader-text">
+          Manage the emails and alerts you receive about your activity
+        </CWText>
+
+        {enableKnockPushNotifications && (
+          <div>
+            <CWText type="h5">Push Notifications</CWText>
+
+            <p>
+              <CWButton
+                label="Subscribe to Push Notifications"
+                onClick={handlePushNotificationSubscription}
+              />
+            </p>
+          </div>
+        )}
+
+        <CWTabsRow>
+          <CWTab
+            label="Community Alerts"
+            isSelected={section === 'community-alerts'}
+            onClick={() => setSection('community-alerts')}
+          />
+          <CWTab
+            label="Subscriptions"
+            isSelected={section === 'subscriptions'}
+            onClick={() => setSection('subscriptions')}
+          />
+        </CWTabsRow>
+
+        {!communityAlerts.isLoading && section === 'community-alerts' && (
+          <>
+            <CWText type="h4" fontWeight="semiBold" className="section-header">
+              Community Alerts
+            </CWText>
+
+            <CWText className="page-subheader-text">
+              Get updates on new threads and discussions from these communities
+            </CWText>
+
+            {Object.entries(bundledSubs)
+              .sort((x, y) => x[0].localeCompare(y[0]))
+              .map(([communityName]) => {
+                const communityInfo = app?.config.chains.getById(communityName);
+
+                if (!communityInfo?.id) return null; // handles incomplete loading case
```

Comment:
this seems unnecessary, as `app.config.chains` gets defined, in the first few API calls after page load. 

Is there a case where this logic gets triggered?